src/interceptors/ClientRequest/index.ts

```diff
@@ -24,19 +24,22 @@ function handleRequest(
   middleware: RequestMiddleware,
   args: any[]
 ): ClientRequest {
+  //The first time we execute this, I'll save the original ClientRequest.
+  //This because is used to restore the dafault one later
   if (!originalClientRequest) {
-    const ClientRequestOverride = createClientRequestOverrideClass(
-      middleware,
-      originalMethod,
-      originalClientRequest
-    )
-
-    debug('patching native http.ClientRequest...')
-
-    // @ts-ignore
-    http.ClientRequest = ClientRequestOverride
+    originalClientRequest = http.ClientRequest
   }
 
+  const ClientRequestOverride = createClientRequestOverrideClass(
+    middleware,
+    originalMethod,
+    originalClientRequest
+  )
+  debug('patching native http.ClientRequest...')
+  //Only http.ClientRequest is overridden because https uses http
+  //@ts-ignore
+  http.ClientRequest = ClientRequestOverride
```

Comment:
When NRI will construct an actual request (non-mocked), it will also bubble to this `handleRequest` function. In that case the flow will be:

```
handleRequest
has original client request? yes
create an override class
patch native "http.ClientRequest" with the override class
create an instance of an override class
```

I think that's not correct. In case of original request this `handleRequest` function must construct an instance of _unpatched_ `http.ClientRequest` class. That's why there's this original client request reference is introduced.

Reply:
When the request is not mocked it is handled by the patched `ClientRequest` but `performOriginalRequest` will be called. `performOriginalRequest` function will call the original http/https.request so our custom instance will only proxy to the normal flow